src/device/mod.rs

```diff
@@ -114,6 +116,7 @@ pub trait Tun: 'static + AsRawFd + Sized + Send + Sync {
     fn new(name: &str) -> Result<Self, Error>;
     #[cfg(not(target_os = "windows"))]
     fn new_from_fd(fd: RawFd) -> Result<Self, Error>;
+    fn new_from_tun_fd(fd: RawFd) -> Result<TunSocket, Error>;
```

Comment:
That is the point of this method, new_from_fd already takes in file descriptor, also int not a good idea to tie trait to concrete implementation of a tait

Reply:
Forgot to do a proper implementation for this workaround. The problem is that new_from_fd() calls ioctl and because the tunnel that is being passed into set_tunnel() is already configured with ioctl, I get a "file already exists" error

Reply:
But we no longer support linux, and this set is only done in linux

Reply:
OK, I noticed now and removed this function completely